tracing/tck/src/main/java/org/eclipse/microprofile/telemetry/tracing/tck/async/MpRestClientAsyncTest.java

```diff
@@ -0,0 +1,140 @@
+
+    @Deployment
+    public static WebArchive createDeployment() {
+
+        ConfigAsset config = new ConfigAsset()
+                .add("otel.bsp.schedule.delay", "100")
+                .add("otel.sdk.disabled", "false")
+                .add("otel.traces.exporter", "in-memory");
+
+        return ShrinkWrap.create(WebArchive.class)
+                .addClasses(InMemorySpanExporter.class, InMemorySpanExporterProvider.class, BasicHttpClient.class,
+                        MpRestClientAsyncTestEndpoint.class)
+                .addAsLibrary(TestLibraries.AWAITILITY_LIB)
+                .addAsServiceProvider(ConfigurableSpanExporterProvider.class, InMemorySpanExporterProvider.class)
+                .addAsResource(config, "META-INF/microprofile-config.properties")
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+
+    private BasicHttpClient basicClient;
+
+    @ArquillianResource
+    URL url;
+
+    public static final String TEST_PASSED = "Test Passed";
+
+    @Inject
+    private InMemorySpanExporter spanExporter;
+
+    @BeforeMethod
+    void setUp() {
+        // Only want to run on server
+        if (spanExporter != null) {
+            spanExporter.reset();
+            basicClient = new BasicHttpClient(url);
+        }
+    }
+
+    @Test
+    public void testIntegrationWithMpRestClient() throws Exception {
+        basicClient.get("/MpRestClientAsyncTestEndpoint/mpclient");
+        readSpans();
+    }
+
+    @Test
+    public void testIntegrationWithMpRestClientAsync() throws Exception {
+        basicClient.get("/MpRestClientAsyncTestEndpoint/mpclientasync");
+        readSpans();
+    }
+
+    public void readSpans() {
+
+        List<SpanData> spanData = spanExporter.getFinishedSpanItems(3);
+
+        List<SpanData> serverSpans = spanExporter.getSpansWithKind(SpanKind.SERVER);
+
+        SpanData firstURL = null;
+        SpanData secondURL = null;
+        SpanData httpGet = spanExporter.getFirst(SpanKind.CLIENT);
+
+        for (SpanData span : serverSpans) {
+            if (span.getAttributes().get(HTTP_TARGET).contains("MpRestClientAsyncTestEndpoint/mpclient")) {
+                firstURL = span;
+            } else {
+                secondURL = span;
+            }
+        }
+        Assert.assertNotNull(firstURL);
+        Assert.assertNotNull(secondURL);
+
+        // Assert correct parent-child links
+        // Shows that propagation occurred
+
+        Assert.assertEquals(httpGet.getSpanId(), secondURL.getParentSpanId());
+        Assert.assertEquals(firstURL.getSpanId(), httpGet.getParentSpanId());
+
+        Assert.assertEquals(HTTP_OK, firstURL.getAttributes().get(HTTP_STATUS_CODE).intValue());
+        Assert.assertEquals(HttpMethod.GET, firstURL.getAttributes().get(HTTP_METHOD));
+        Assert.assertEquals("http", firstURL.getAttributes().get(HTTP_SCHEME));
+
+        Assert.assertEquals("HTTP GET", httpGet.getName());
```

Comment:
Do we really want a span name without the target path?

Reply:
No, the [semantic conventions](https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/http/#name) don't require the span name to be anything in particular, so we shouldn't test for it. See also #86

Reply:
It actually does: https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/http/#name

Reply:
Right, but it's only a SHOULD, not a MUST, which is why I thought we shouldn't be testing it here.

Reply:
Please remove the assert on the span name.